#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("bin_size", int_t, 0, "bin size of color histogram", 20, 1, 256)
gen.add("queue_size", int_t, 0, "queue size of message", 100, 1, 1000)
gen.add("white_threshold", double_t, 0, "white threshold", 0.1, 0.0, 1.0)
gen.add("black_threshold", double_t, 0, "black threshold", 0.1, 0.0, 1.0)

histogram_policy_enum = gen.enum([gen.const("HUE", int_t, 0, "use hue only"),
                                  gen.const("HUE_AND_SATURATION", int_t, 1, "use hue and saturation")],
                                 "histogram policy")
gen.add("histogram_policy", int_t, 0, "histogram policy", 0, 0, 1,
        edit_method=histogram_policy_enum)

exit(gen.generate(PACKAGE, PACKAGE, "ColorHistogram"))
